priam/src/main/java/com/netflix/priam/backup/AbstractBackup.java

```diff
@@ -100,7 +100,9 @@ public AbstractBackupPath retriableCall() throws Exception {
 
                 addToRemotePath(abp.getRemotePath());
             } catch (Exception e) {
-                logger.error("Failed to upload local file {} within CF {}. Ignoring to continue with rest of backup.", file.getCanonicalFile(), parent.getAbsolutePath(), e);
+                //Throw exception to the caller. This will allow them to take appropriate decision.
+                logger.error("Failed to upload local file {} within CF {}.", file.getCanonicalFile(), parent.getAbsolutePath(), e);
```

Comment:
Should we throw the exception or just collect all the failed files and throw one exception at the end after uploading all successful files?

Reply:
I had a discussion around this as to what is the strategy in case of failure. How will upload of other remaining objects as part of "snapshot" help because that snapshot will never be marked as "successful" and thus can not be used for "restore" path either? 
The current way to come out of snapshot failure is to actually go and take a complete new snapshot. Throwing exception right away will ensure we fail-fast rather than fail and notify later.

Reply:
Ok, I'm +1 if with resumable snapshot of course :-)

Reply:
Yes, fail-fast with resumable is the way to go!